### PR TITLE
Create CONTRIBUTING.md file

### PR DESCRIPTION
**Pull request by waldyrious**
- Add CONTRIBUTING.md file
- Format markdown files to break long lines and improve wording

For future reference, this PR was prompted by [this discussion](https://github.com/hamstar/Wikimate/pull/133#discussion_r697483814) in #133.

### CODE REVIEW

File: CONTRIBUTING.md [hunk 1]
```diff
@@ -0,0 +1,34 @@
...
+   and allows the author of each change to properly summarize it.
+4. **Update the documentation in `USAGE.md` if necessary**.
+   When writing prose, break lines [semantically](https://rhodesmill.org/brandon/2012/one-sentence-per-line/).
+   We don't have a strict maximum line length
+   (especially since things like long URLs can easily surpass them),
+   but you should start thinking about breaking lines once they're over approaching 100 characters in length.
+
+## PHP coding guidelines
+
+- Use a `\` before standard class names
```
Comment (waldyrious) [inline CONTRIBUTING.md:34]: @Xymph this entry is just a placeholder for this section. Do you think there are specific recommendations we could add here? Or should we simply link to an existing PHP style guide or best practices document?

File: CONTRIBUTING.md [hunk 2]
```diff
@@ -0,0 +1,34 @@
...
+   We recommend [these principles](https://chris.beams.io/posts/git-commit/#seven-rules)
+   for writing great commit messages (but they're not enforced, so don't sweat it 🙂).
+3. **Include a `CHANGELOG.md` entry in every pull request**.
+   This makes it much easier to prepare releases,
+   and allows the author of each change to properly summarize it.
+4. **Update the documentation in `USAGE.md` if necessary**.
+   When writing prose, break lines [semantically](https://rhodesmill.org/brandon/2012/one-sentence-per-line/).
+   We don't have a strict maximum line length
+   (especially since things like long URLs can easily surpass them),
+   but you should start thinking about breaking lines once they're over approaching 100 characters in length.
```
Comment (waldyrious) [inline CONTRIBUTING.md:30]: Oops, noticed a typo here:
```suggestion
   but you should start thinking about breaking lines once they're approaching 100 characters in length.
```

File: CONTRIBUTING.md (hunk 1 above)
Comment (waldyrious) [reply CONTRIBUTING.md:34]: For historical record, this came up in https://github.com/hamstar/Wikimate/pull/133#issuecomment-907209101.

Comment (Xymph) [reply CONTRIBUTING.md:34]: The main standards are the [PSRs](https://www.php-fig.org/psr/), and I've looked into some occasionally but am not well versed in any, and am not sure Wikimate even meets the basic coding standard in all relevant aspects.

The `\` specifies [global namespace](https://www.php.net/manual/en/language.namespaces.global.php), but as said, we don't (yet) use a namespace so I don't think it requires highlighting at this point.
I am simply not very experienced with namespaces, autoloading, composer, and similar modern technologies (although I worked extensively with Zend Framework 1, but that was during 2010-2014), so not in a strong position to recommend a style guide or best practices doc.

Comment (waldyrious) [reply CONTRIBUTING.md:34]: Ok, let me remove this section for now, and maybe later on we can add one if we feel it's warranted.

File: CONTRIBUTING.md (hunk 2 above)
Comment (waldyrious) [reply CONTRIBUTING.md]: Fixed with the [latest push](https://github.com/hamstar/Wikimate/compare/2206bf4..f0cd8f9).

File: CONTRIBUTING.md (hunk 1 above)
Comment (waldyrious) [reply CONTRIBUTING.md]: Done in the [latest push](https://github.com/hamstar/Wikimate/compare/2206bf4..f0cd8f9).

**Comment by Xymph** [comment]
No remarks on contributing/governance/changelog.

In Usage, consider linefeeds at [58](https://github.com/hamstar/Wikimate/blame/f0cd8f90b7a499593b4cb651c0ec16a6f0548b39/USAGE.md#L58), [87](https://github.com/hamstar/Wikimate/blame/f0cd8f90b7a499593b4cb651c0ec16a6f0548b39/USAGE.md#L87), [185](https://github.com/hamstar/Wikimate/blame/f0cd8f90b7a499593b4cb651c0ec16a6f0548b39/USAGE.md#L185), [311](https://github.com/hamstar/Wikimate/blame/f0cd8f90b7a499593b4cb651c0ec16a6f0548b39/USAGE.md#L311);
and in Readme at [10](https://github.com/hamstar/Wikimate/blame/f0cd8f90b7a499593b4cb651c0ec16a6f0548b39/README.md#L10)-11, [19](https://github.com/hamstar/Wikimate/blame/f0cd8f90b7a499593b4cb651c0ec16a6f0548b39/README.md#L19)-20 and more (looks like you skipped it entirely).

**Comment by waldyrious** [comment]
Great suggestions for USAGE.md. I've applied them all. Same in the README. I hadn't really looked into lines that might need re-wrapping; I only handled lines that were already relatively long. I have now tried to locate other instances where re-wrapping might make sense.

As for lines 10-11 in the README, I'm not sure what wrapping would work better in your view (without breaking the sentence's clauses apart). I've moved the second sentence to its own line, but I can't think of a better way to wrap the first sentence. If anything, I could make it longer, by moving the "is v0.15.0," to the first line. Is this what you had in mind?

File: USAGE.md [hunk 3]
```diff
@@ -214,7 +218,8 @@ echo $file->getAspectRatio();
 
 #### Downloading...
 
-You can obtain the data of the file by using the `downloadData()` method and use it in your script, or write it directly to a local file via the `downloadFile()` method.
+You can obtain the data of the file by using the `downloadData()` method,
```
Comment (Xymph) [inline USAGE.md:221]: A comma shouldn't be added here: either the file data from `downloadData` is used in the script, or the file data is written out via `downloadFile`. The comma suggests that after the first method, one still has a choice.

File: README.md [hunk 4]
```diff
@@ -8,22 +8,23 @@ It currently consists of three classes:
 * **WikiFile** – Provides an interface to downloading/uploading files and getting their properties.
 
 The [latest released version](https://github.com/hamstar/Wikimate/releases) of Wikimate
-is v0.15.0, released on Aug 26, 2021. It requires MediaWiki v1.27 or newer.
+is v0.15.0, released on Aug 26, 2021.
```
Comment (Xymph) [inline README.md:11]: This is what I had in mind, because this is the only clause to change every release.

**Review by Xymph — CHANGES_REQUESTED** [review]
Almost there! :smiley:

File: USAGE.md (hunk 3 above)
Comment (waldyrious) [reply USAGE.md:221]: Ah, that's a tricky construction indeed. I'll keep the "and..." part in this line to help drive home the point that this is meant to be a sequence. Maybe we can shorten the clause a bit:

> You can get the file content with the `downloadData()` method and use it in your script,

WDYT?

Comment (Xymph) [reply USAGE.md:221]: Then the second part implicitly becomes: "write the file content to a local file via"... which is weird. That's why I used "the data of the file" phrase. I'd just stick with the original two-line sentence.

Comment (waldyrious) [reply USAGE.md]: Good point. [Done](https://github.com/hamstar/Wikimate/compare/e7f2f232a9c243b8ac3a7c6017562e80e165b511..656e027da9b033080894cc7bc0c4d391cc598389).

**Review by Xymph — APPROVED** [review]

**Comment by Xymph** [comment]
And that's a wrap! :+1:

**Pull request merged**